src/context/videos/Context.jsx

```diff
@@ -1,16 +1,51 @@
-import { createContext, useContext, useReducer } from "react";
+import {
+  createContext,
+  useContext,
+  useReducer,
+  useState,
+  useEffect,
+} from "react";
+import axios from "axios";
+import { filterByCategory, getResultantVideos } from "./utils";
 import { videosReducer } from "./Reducer";
 
 const VideosContext = createContext();
 
 const VideosProvider = ({ children }) => {
   const [videoState, videoDispatch] = useReducer(videosReducer, {
-    categories: [],
+    category: "True Sight",
   });
 
+  const [videos, setVideos] = useState([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get("/api/videos");
+        if (response.status === 200) {
+          setVideos(response?.data?.videos);
+        }
+      } catch (e) {
+        console.error(e);
+      }
```

Comment:
remove console, can print some toast message in error

Reply:
Not final version will fix.